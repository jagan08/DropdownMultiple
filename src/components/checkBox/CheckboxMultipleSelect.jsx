import React, {Component} from 'react';
import './checkBoxStyle.css';

class CheckboxMultipleSelect extends Component {  
  /**
   * @description Render the check box item in dropdown
   */
  render () {
    const {item} = this.props;
    return (
      <span>
        <input
          type="checkbox"
          defaultChecked={item.selected}
          className="regular-checkbox big-checkbox"
          onClick={() => {
            this.props.toggleItem (item.id, item.key);
          }}
          key={item.title}
        />
        <li className="dd-list-item" key={item.id}>
          {item.title} {item.selected}
        </li>
      </span>
    );
  }
}

export default CheckboxMultipleSelect;
