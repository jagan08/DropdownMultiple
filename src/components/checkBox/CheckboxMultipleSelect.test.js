import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import CheckboxMultipleSelect from './CheckboxMultipleSelect';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});


describe("CheckboxMultipleSelect Componenet", () => {
    const item = {
        id: 0,
        title: 'Jacks or Better Double up',
        selected: false,
        key: 'list'
    };
  test("it should render the app", () => {
    const wrapper  = shallow(<CheckboxMultipleSelect item={item}/>); 
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();    
  })

})


