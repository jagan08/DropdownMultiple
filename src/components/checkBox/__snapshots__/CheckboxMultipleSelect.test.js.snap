// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Componenet it should render the app 1`] = `ShallowWrapper {}`;
